python/test/unit/spacewalk/satellite_tools/test_reposync.py

```diff
@@ -212,6 +212,63 @@ def test_sync_success_regen(self):
                          ((["Label"], [], "server.app.yumreposync"), {}))
         self.assertEqual(self.reposync.taskomatic.add_to_erratacache_queue.call_args,
                          (("Label", ), {}))
+    
+    def _mock_cfg(self) -> Mock:
+        CFG = Mock()
+        CFG.MOUNT_POINT = '/tmp'
+        CFG.PREPENDED_DIR = ''
+        CFG.AUTO_GENERATE_BOOTSTRAP_REPO = 1
+        return CFG
+    
+    def _mock_repo_plugin(self, pkg_list) -> Mock:
+        plug = Mock()
+        plug.list_packages = Mock(return_value=pkg_list)
+        plug.num_packages = len(pkg_list)
+        plug.num_excluded = 0
+        plug.repo.pkgdir = "/tmp"
+        return plug
+
+    @patch("uyuni.common.context_managers.initCFG", Mock())
+    @patch("spacewalk.satellite_tools.reposync.log2", Mock())
+    @patch("spacewalk.satellite_tools.reposync.os", os)
+    @patch("spacewalk.satellite_tools.reposync.ThreadedDownloader")
+    @patch("spacewalk.satellite_tools.reposync.multiprocessing.Pool")
+    def test_import_packages_excludes_failed_pkgs(self, pool, downloader):
+        """
+        When downloader fails to download a subset of packages
+        Then the RepoSync.import_packages function should not process the failed packages
+        """
```

Comment:
I find this test a bit hard to follow. It's not obvious what's the system under test (SUT) and what are the collaborators (that can be substituted by test doubles). Could you maybe re-structure this test a bit and maybe add a few comments?

Reply:
It is absolutely hard to follow. Honestly, the function under test is hard to follow as well :/ It does multiple things with multiple objects that we do not care about in this particular test, hence the number of patches.

I have added a few comments and refactored the test to make it just a little bit less noisy. However, the main collaborators should be objects injected by pytest into the method parameter list, right?

Reply:
You're totally right, `import_packages` is doing too much.

Thanks for adding the comments, I think the test is better now. It's not pretty but following the function, it's understandable.